COMDOX-1734: Updated SCP topics for clarity (#941)
commerce-docs/microsite-commerce-storefront: src/content/docs/get-started/backends.mdx

Reviewer: Use Commerce on cloud or on premises instead of "Commerce PaaS" throughout these topics.

Reviewer: For header:  Adobe Commerce - insert line break 
(on cloud, on premises)

---
title: Backend options
description: Identify your Adobe Commerce backend type (PaaS, Adobe Commerce as a Cloud Service, or Adobe Commerce Optimizer) and confirm prerequisites before storefront setup.
---

import Aside from '@components/Aside.astro';
import CardGrid from '@components/CardGrid.astro';
import Diagram from '@components/Diagram.astro';
import Link from '@components/Link.astro';
import LinkCard from '@components/LinkCard.astro';
import Options from '@components/Options.astro';
import Option from '@components/Option.astro';
import TableWrapper from '@components/TableWrapper.astro';
import Term from '@components/Term.astro';

> The storefront supports a few backend types, each with its own prerequisites, so follow only the path that matches your project. Choosing the wrong type leads to misleading signals—GitHub and DA.live look healthy while previews or drop-ins fail—and results in installing packages your license does not require.

Reviewer 2: ```suggestion
The storefront supports a few backend types, each with its own prerequisites, so follow only the path that matches your project. Choosing the wrong type leads to misleading signals. For example, GitHub and DA.live look healthy, but previews or drop-ins fail. As a result, you might install packages your license does not require.
```

For authoring storefront pages (Document Authoring, Universal Editor), see [Document Authoring Quick Start](/merchants/quick-start/document-authoring/). This page covers the Commerce backend the storefront connects to.

> Drop-in components—B2C (shopper experiences) and B2B (company accounts, quotes, purchase orders)—require an Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer license. PaaS customers who add an Adobe Commerce Optimizer license get drop-in support as part of that license. There is no separate "B2B backend" type.

Reviewer 2: ```suggestion
All drop-in components require an Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer license. Adobe Commerce on Cloud and on-premises customers who add an Adobe Commerce Optimizer license get drop-in support as part of that license. There is no separate "B2B backend" type.
```

I hate the em-dashed asides that the AI loves so much.

> Drop-in components — B2C (business-to-consumer, meaning typical shopper experiences) and B2B (business-to-business, including company accounts, negotiable quotes, and purchase orders) — require an Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer license. PaaS customers who add an Adobe Commerce Optimizer license are covered, and that license includes drop-in usage. There is no separate “B2B backend” type.

Reviewer 2: Rewrite to avoid "PaaS"

Backends fall into the following categories:

- Commerce on Cloud or on-premises — Existing Adobe Commerce on Cloud infrastructure or on-premises deployment without an Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer license
- Adobe Commerce as a Cloud Service — Cloud Service deployment whose license includes Storefront
- Adobe Commerce Optimizer — New Adobe Commerce Optimizer deployment, or an existing PaaS deployment that added Adobe Commerce Optimizer

<Aside type="tip" title="Related documentation">
This page covers Commerce backends only. For Adobe Commerce product help (Admin, deployments, merchandising services, integrations), start from <Link href={"https://experienceleague.adobe.com/en/docs/commerce"} text={"Adobe Commerce documentation"} /> on Experience League.

> For pages and blocks on da.live, see <Link href={"https://docs.da.live/"} text={"Document Authoring documentation"} /> or [Document Authoring Quick Start](/merchants/quick-start/document-authoring/).

Reviewer 2: Is it DA.live or da.live? An earlier topic in this PR had it DA.live

</Aside>

</Option>

</Options>

## PaaS: required packages and services

Install and configure the following on Commerce on Cloud or on-premises before you create your storefront. Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer manage these automatically, so skip this section if you are not on Commerce on Cloud or on-premises.

### Minimum checklist before you create a storefront

Before <Term>Site Creator</Term>, manual configuration, or local preview can rely on Commerce APIs, confirm these items on Commerce on Cloud and on-premises:

1. You are on Adobe Commerce v2.4.8 or later (Magento Open Source is not supported for this storefront path).
1. You installed the <Term>Storefront Compatibility package</Term>, which fills gaps in the core Commerce GraphQL schema that drop-ins require. See [Storefront Compatibility Package](/setup/configuration/storefront-compatibility/) for who installs it, or [Manual installation](/setup/configuration/storefront-compatibility/install/) if you are on Commerce on Cloud or on-premises with Adobe Commerce Optimizer.

> 1. If you build a B2B storefront, install the Storefront Compatibility B2B Package (SCP-B2B) on the same Commerce instance, after Adobe Commerce B2B is already there.

Reviewer 2: B2B is installed automatically. It'll need to be enabled.

|---------|---------------|-----------------------------------|--------------------------|
| License includes Storefront | No | Yes | Yes |
| Setup | Manual install required | Fully automated | Fully automated |

> | Version | v2.4.7+ (self-managed Commerce) | Commerce managed by Adobe (SaaS) | Optimizer managed by Adobe (SaaS) |

Reviewer 3: Do we need to update this, since we are planning to keep SCP-4.7?

@tpogrebniak, we are making the SCP-4.7 package with internal and external availability groups. Do we need to remove external and add ACO with the availability group for the new SCP-4.7.x build?

Author: @deepak-soni1 and @tpogrebniak, I made no 2.4.7 changes in my lastest updates based on your most recent review. I'm waiting for others to answer your 2.4.7 questions.

> | Version | v2.4.8+ (self-managed Commerce) | Commerce managed by Adobe (SaaS) | Optimizer managed by Adobe (SaaS) |

Reviewer 2: GH won't let me leave a comment on the table header, but you should not be using the term Commerce PaaS. We need to wean ourselves off of using PaaS and SaaS. 

```suggestion
| Version | v2.4.8+ (self-managed Commerce) | Versionless | Optimizer managed by Adobe (SaaS) |
```

I don't know what to do with "Optimizer managed by Adobe (SaaS)". Ask Margaret

Edit: The licensing page says that dropins are supported only for ACCS and Optimizer customers. Why are you have that Commerce PaaS column in the first place?

Reviewer: since this is about the version -- it would be  N/A for Commerce as a Cloud Service and Commerce Optimizer:

"Not Applicable (SaaS Model). Adobe Commerce as a Cloud Service / Commerce Optimizer utilizes a cloud-native, continuous-release architecture. Updates are delivered as rolling, fully managed upgrades handled directly by Adobe rather than static, point-in-time version releases.Hosted service"

## Backend topology

> This diagram focuses on the backend layer. It adds two optional pieces you might use during a phased move: <Term>Luma Bridge</Term> and the Adobe Commerce Optimizer Connector. Luma is Adobe's classic server-side Commerce storefront theme. If you run a <Term>Luma</Term> storefront today and are moving to Edge Delivery Services, Luma Bridge is a PHP module on Commerce that enables Luma pages and EDS drop-ins to share the same cart and sign-in session during the transition. For the full stack (drop-ins, boilerplate, and how services connect), see [Commerce services and backends](/get-started/architecture/commerce-services-and-backends/).

Reviewer: phased move ==> phased migration to Adobe Commerce as a Cloud Service or Adobe Commerce Optimizer.

#### Version

> v2.4.7 or later

Reviewer 3: Do we need to update 2.4.7 ?

> Before <Term>Site Creator</Term>, manual configuration, or local preview can rely on Commerce APIs, confirm these items on Commerce PaaS:

Reviewer 2: ```suggestion
Before <Term>Site Creator</Term>, manual configuration, or local preview can rely on Commerce APIs, confirm these items on Commerce on Cloud and on-premises:
```

1. You are on Adobe Commerce v2.4.8 or later (Magento Open Source is not supported for this storefront path).

> 1. You installed the <Term>Storefront Compatibility package</Term> ([Storefront Compatibility Package](/setup/configuration/storefront-compatibility/) — [Manual installation](/setup/configuration/storefront-compatibility/install/) on Commerce PaaS or on-premises with Adobe Commerce Optimizer), which fills gaps in the core Commerce GraphQL schema that drop-ins require.

Reviewer 2: ```suggestion
1. You installed the <Term>Storefront Compatibility package</Term> ([Storefront Compatibility Package](/setup/configuration/storefront-compatibility/) — [Manual installation](/setup/configuration/storefront-compatibility/install/) on Commerce on Cloud or on-premises with Adobe Commerce Optimizer), which fills gaps in the core Commerce GraphQL schema that drop-ins require.
```

> 1. If you build a B2B storefront, you installed the Storefront Compatibility B2B Package (SCP-B2B) on the same Commerce instance ([B2B Compatibility Package](/setup/configuration/storefront-compatibility/b2b/) — [Manual installation](/setup/configuration/storefront-compatibility/install/#install-the-b2b-package) on Commerce PaaS or on-premises with Adobe Commerce Optimizer). Adobe Commerce B2B must be on the instance before you install SCP-B2B.

Reviewer 2: ```suggestion
1. If you build a B2B storefront, you installed the Storefront Compatibility B2B Package (SCP-B2B) on the same Commerce instance ([B2B Compatibility Package](/setup/configuration/storefront-compatibility/b2b/) — [Manual installation](/setup/configuration/storefront-compatibility/install/#install-the-b2b-package) on Commerce on Cloud or on-premises with Adobe Commerce Optimizer). Adobe Commerce B2B must be on the instance before you install SCP-B2B.
```

I'm not going to suggest changes for "Commerce PaaS" any more. Do a global search and replace for that string

1. You connected the storefront services your project needs (at minimum, plan for Services Connector and Catalog Service, then add Live Search, Product Recommendations, and Data Connection per your rollout). Use the [service documentation table](#storefront-services) below for links.

### Storefront Compatibility package

> See [Storefront Compatibility Package](/setup/configuration/storefront-compatibility/) for who installs the package on each backend. On Commerce PaaS or on-premises with Adobe Commerce Optimizer, install it with Composer on your Commerce server. Your storefront repo still uses npm for JavaScript drop-ins, but these modules run as PHP on the Commerce server. See [Manual installation](/setup/configuration/storefront-compatibility/install/) to pick the version that matches your Adobe Commerce version, and read [Commerce services](/get-started/architecture/commerce-services-and-backends/#commerce-services) for how Commerce services fit together.

Reviewer 2: Consider whether this is duplication of what's in SCP index page
